### PR TITLE
PR: Add more file type icons to File Explorer

### LINKED ISSUE #3825: Improve icons in the File explorer
Icons in the File explorer aren't anymore populated/updated

**What steps will reproduce the problem?**

1. Fresh install WinPython 3.5.2.3QT5 
2. open Spyder, 
3. activate the File Explorer

I get that : 

but in previous version, I had that (icons are the same as in window explorer)

* Spyder Version: 3.1.0.dev0
* Python Version: 3.5
* Operating system: Win 7 enterprise

Winpython maintainer (stonebig) did reproduce the problem, see issue : https://github.com/winpython/winpython/issues/446#issuecomment-266191173

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #3825 

----

Icons are based on font-awesome's ``file-*-o`` icons and they are selected based on the inferred MIME type of each file:

captura de pantalla de 2017-06-13 00-16-24

### CODE REVIEW

**Comment by ccordoba12** [comment]
Great work @andfoy! I just have one question. On systems other than Windows, I'd like to report the icon of the file a symlink points to. Is this covered in your PR?

@dalthviz, please test this PR and see if it works on Windows correctly.

**Comment by andfoy** [comment]
@ccordoba12 I'll check if the icons apply to symlinks, if not, then they will be added! :-)

**Comment by ccordoba12** [comment]
Thanks a lot @andfoy!

**Comment by andfoy** [comment]
@ccordoba12 Symlinks preserve parent's icon only if the name preserves the original extension, should I override this behaviour?

**Comment by ccordoba12** [comment]
No, let's leave things as they are for now as see what users say about this.

We're waiting for @dalthviz to see if this works well on Windows.

**Comment by dalthviz** [comment]
@ccordoba12 @andfoy as soon as I manage to fix my Windows installation I will test this :+1:

**Comment by dalthviz** [comment]
@andfoy @ccordoba12 some of the icons changed 👍 :
 
imagen

but there is an error:

imagen

I think that instead of `AudioFileIcon` the correct key is `SoundFileIcon`

**Comment by ccordoba12** [comment]
Also, it's not detecting `xlsx` as Excel files, and probably the same happens for `docx` and `pptx`.

File: spyder/widgets/explorer.py [hunk 1]
```diff
@@ -123,7 +140,23 @@ def icon(self, icontype_or_qfileinfo):
...
-                return ima.icon('FileIcon')
+                basename = osp.basename(fname)
+                mime_type, _ = mime.guess_type(basename)
+                icon = ima.icon('FileIcon')
+                if mime_type is not None:
+                    file_type, bin_name = mime_type.split('/')
+                    if file_type == 'text':
+                        icon = ima.icon('TextFileIcon')
+                    elif file_type == 'audio':
+                        icon = ima.icon('AudioFileIcon')
```
Comment (ccordoba12) [inline spyder/widgets/explorer.py]: @dalthviz is right, this must be `SoundFileIcon`.

Comment (ccordoba12) [inline spyder/widgets/explorer.py]: Although I think it's better to leave it as `AudioFileIcon` and change it in `icon_manager`.

**Comment by andfoy** [comment]
Actually, it seems that ``mimetypes`` modules is not recognizing docx, xlsx and pptx document extensions:
captura de pantalla de 2017-06-13 14-43-42

**Comment by ccordoba12** [comment]
Then force the association using the file extension, please.

**Comment by ccordoba12** [comment]
@dalthviz, please run a final test for this one, please.

**Comment by dalthviz** [comment]
@andfoy @ccordoba12 testing this is not working, but it's just because a missing dot in the dic that has the MS Office extensions (i.e `.xlsx`, `.docx`, `.pptx`). Also in the line `151` the icon assigment should be something like `icon = ima.icon(self.OFFICE_FILES[extension])`

**Comment by ccordoba12** [comment]
@dalthviz, please give it another try (hopefully the last one :-)

**Comment by dalthviz** [comment]
@andfoy @ccordoba12 👍 

imagen

**Pull request merged**